frontend/src/components/AuthenticationGuard.tsx

```diff
@@ -0,0 +1,19 @@
+import { withAuthenticationRequired } from "@auth0/auth0-react";
+
+
+
+// This component takes another React component as  props and checks if the user is authenticated. If so, use that route. Otherwise, the user will be ridercet to the auth0 login page.
+
+interface ComponentProps {
+    componentAsProps:  React.ComponentType<any>
+}
+export const AuthenticationGuard: React.FC<ComponentProps> = ({componentAsProps}) => {
```

Comment:
For cleaner code lets import `FC` from React so we dont need to use dot notation

Reply:
And for convention add spaces between curly braces like `({ componentAsProps })`